### PR TITLE
Add check on dimensions in Gemm opset 6

### PR DESCRIPTION
**Pull request by gramalingam**
Fix https://github.com/onnx/onnx/issues/6185

### CODE REVIEW

File: onnx/defs/math/old.cc
Comment (justinchuby) [inline onnx/defs/math/old.cc:3313]: I would include the actual shape in the error message to help diagnose issues.

**Review by justinchuby — APPROVED** [review]

File: onnx/defs/math/old.cc
Comment (gramalingam) [reply onnx/defs/math/old.cc]: Done, by calling utility which does that.

**Comment by gramalingam** [comment]
@postrational : this needs an operator SIG approval ... would be great if you could take a look. I can't approve it myself.

**Review by justinchuby — APPROVED** [review]

**Pull request merged**
